src/System.CommandLine.Subsystems/Subsystems/Annotations/AnnotationStorageExtensions.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.CommandLine.Subsystems.Annotations;
+
+/// <summary>
+/// Handles storage of annotations associated with <see cref="CliSymbol"/> instances.
+/// </summary>
+public static partial class AnnotationStorageExtensions
```

Comment:
Does this class need to be public?

Reply:
Yes. Implementors of subsystems need to use it to store any newly defined annotations. However, it's in the `System.CommandLine.Subsystems.Annotations` so it's not something most users will see.